source/parser/parser_expressions.cpp

```diff
@@ -1222,16 +1222,22 @@ void Parser::Parse_Num_Factor (EXPRESS& Express,int *Terms)
                         // JN2007: Image map dimensions:
                         CASE (PIGMENT_ID_TOKEN)
                             Pigment = reinterpret_cast<PIGMENT *>(Token.Data);
-                            if(Pigment->Type != BITMAP_PATTERN)
+                            if((Pigment->Type != BITMAP_PATTERN) && (Pigment->Type != DENSITY_FILE_PATTERN))
```

Comment:
While you're at it, can you do me a favour and rearrange the structure from `if (!A && !B) { ... } else if (A) { ... } else if (B) { ... }` to `if (A) { ... } else if (B) { ... } else { ... }`?